getting_started/build-guide/build-with-zephyr.rst

```diff
@@ -0,0 +1,92 @@
+.. _build-with-zephyr:
+
+Build SOF with `Zephyr <https://zephyrproject.org/>`__
+#####################
+
+.. contents::
+   :local:
+   :depth: 3
+
+This guide describes the process of building and running |SOF| as a Zephyr
+application.
+
+.. note::
+
+    The example uses ``$ZEPHYR_WORKSPACE`` as the working directory.
+
+Preparation
+***********
+
+The easiest way to build Zephyr is by using its recommended toolchain. Follow
+instructions in the
+`Install a Toolchain <https://docs.zephyrproject.org/latest/getting_started/index.html#install-a-toolchain>`__
+guide for toolchain installation details.
+
+Check out and build
+*******************
+
+Zephyr uses ``west`` as a source management and building system. Follow the
+Zephyr
+`Getting Started <https://docs.zephyrproject.org/latest/getting_started/index.html#>`__
+guide for dependencies and for ``west`` installation.
+
+Initialize a new ``west`` repository
+
+.. code-block:: bash
+
+    mkdir $ZEPHYR_WORKSPACE
+    cd $ZEPHYR_WORKSPACE
+    west init
+    west update
+
+.. note::
+
+    ``west init`` with no additional parameters configures a new repository with
+    the default Zephyr repository and branch. However, at the moment the SOF
+    integration into Zephyr isn't yet complete, therefore an alternative
+    repository has to be cloned:
+
+.. code-block:: bash
+
+    west init --manifest-url https://github.com/andyross/zephyr --manifest-rev sof-rebased-22Oct20
+
+    Some additional work is also required in the SOF directory:
+
+.. code-block:: bash
+
+    pushd modules/audio/sof
+    git clone --recurse-submodules https://github.com/thesofproject/rimage.git
+    mkdir rimage/build
+    cd rimage/build
+    cmake ..
+    make
+    sudo make install
```

Comment:
I never had to install `rimage`, this is weird.

Maybe also a short warning that Zephyr has its own, duplicate copy of rimage that is not used right now? https://github.com/thesofproject/sof/issues/3517

Reply:
> I never had to install `rimage`, this is weird.

@marc-hb I think SOF uses rimage directly from its build directory, whereas Zephyr calls rimage from its signing code by searching standard executable paths or using a command-line parameter

Reply:
I just saw that `west sign` has a `-p, --tool-path` option. Can you try it and mention it as an alternative to `sudo`? 

AFAIK neither Zephyr or SOF ever required root access to build anything (as they should) so this single `sudo` line would tear all that down. Security concerns aside, root access is a blocker for people using (powerful) shared systems - and you DO need powerful systems if you want to perform any kind of serious `sanitycheck` testing.